### PR TITLE
Update GamePasses.lua

### PR DESCRIPTION
**Pull request by howmanysmall**
pcall UserOwnsGamePassAsync.

### CODE REVIEW

File: Modules/GamePasses.lua [hunk 1]
```diff
@@ -12,30 +12,37 @@ local boughtGamePasses = WeakInstanceTable()
...
+MarketplaceService.PromptGamePassPurchaseFinished:Connect(function(player, gamePassId, purchased)
 	if not purchased then return end
 	boughtGamePasses[player] = boughtGamePasses[player] or {}
 	boughtGamePasses[player][gamePassId] = true
 end)
 
 local GamePasses = {}
 
+local function userOwnsGamePassAsync(player, gamePassId)
+	local success, doesOwn = pcall(MarketplaceService.UserOwnsGamePassAsync, MarketplaceService, player.UserId, gamePassId)
```
Comment (Kampfkarren) [inline Modules/GamePasses.lua]: It's optimal, but it's unreadable, would rather this just be pcalling an anonymous function.

Comment (Kampfkarren) [inline Modules/GamePasses.lua]: It's optimal, but it's unreadable, would rather this just be pcalling an anonymous function.

File: Modules/GamePasses.lua [hunk 2]
```diff
@@ -12,30 +12,37 @@ local boughtGamePasses = WeakInstanceTable()
...
+	return success and doesOwn or false
```
Comment (Kampfkarren) [inline Modules/GamePasses.lua]: `or false` isn't necessary, if `success` is false, then `false and nil` -> `false`

File: Modules/GamePasses.lua [hunk 3]
```diff
@@ -12,30 +12,37 @@ local boughtGamePasses = WeakInstanceTable()
...
 	end
 
 	if RunService:IsServer() then
-		Players.PlayerAdded:connect(checkGamePassOwnership)
-		for _, player in pairs(Players:GetPlayers()) do
-			coroutine.wrap(checkGamePassOwnership)(player)
+		Players.PlayerAdded:Connect(checkGamePassOwnership)
+		for _, player in ipairs(Players:GetPlayers()) do
+			local thread = coroutine.create(checkGamePassOwnership)
+			coroutine.resume(thread, player)
```
Comment (Kampfkarren) [inline Modules/GamePasses.lua]: Why did you change this from wrap? It's more readable IMO.

Comment (howmanysmall) [reply Modules/GamePasses.lua]: It is more readable, I probably changed it for either the fact that `coroutine.resume` returns success and error like `pcall` and it's a bit faster.

Comment (Kampfkarren) [reply Modules/GamePasses.lua]: We're not using the success/error here, and it would probably not work well here anyway since UserOwnsGamePassAsync yields

**Comment by Kampfkarren** [comment]
Stale.
